torch/distributions/laplace.py

```diff
@@ -0,0 +1,39 @@
+from numbers import Number
+import torch
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+class Laplace(Distribution):
+    r"""
+    Creates a Laplace distribution parameterized by `loc` and 'scale'.
+
+    Example::
+
+        >>> m = Laplace(torch.Tensor([0.0]), torch.Tensor([1.0]))
+        >>> m.sample()  # Laplace distributed with loc=0, scale=1
+         0.1046
+        [torch.FloatTensor of size 1]
+
+    Args:
+        loc (float or Tensor or Variable): mean of the distribution
+        scale (float or Tensor or Variable): scale of the distribution
+    """
+    has_rsample = True
+
+    def __init__(self, loc, scale):
+        self.loc, self.scale = broadcast_all(loc, scale)
+        if isinstance(loc, Number) and isinstance(scale, Number):
+            batch_shape = torch.Size()
+        else:
+            batch_shape = self.loc.size()
+        super(Laplace, self).__init__(batch_shape)
+
+    def rsample(self, sample_shape=torch.Size()):
+        shape = self._extended_shape(sample_shape)
+        u = self.loc.new(*shape).uniform_() - .5
+        return self.loc - self.scale * u.sign() * torch.log(1 - 2 * u.abs())

```

Comment:
I believe `.uniform_()` samples in the the half-open interval `[0,1)`, so that this will occasionally return +INF. I'm not sure the best way around this, but maybe a Bernoulli + Exponential or a different trick using `.sign()`?